dotCMS/src/main/java/com/dotcms/rest/api/v1/taillog/TailLogResource.java

```diff
@@ -0,0 +1,214 @@
+        String name = null;
+        for (int i = 0; i < 1000; i++) {
+            name = "LogTailer" + i + ":" + fileName;
+            Thread t = ThreadUtils.getThread(name);
+            if (t == null) {
+                break;
+            }
+            if (i > 100) {
+                sendError("Too many Logger threads");
+            }
+        }
+
+        thread.setName(name);
+
+        thread.start();
+
+        return thread.getEventOutput();
+    }
+
+    private EventOutput sendError(final String errorMessage){
+
+        final EventOutput eventOutput = new EventOutput();
+
+        final OutboundEvent.Builder eventBuilder = new OutboundEvent.Builder();
+        eventBuilder.name("failure");
+        eventBuilder.data(Map.class,
+                map("failure", errorMessage));
+        final OutboundEvent event = eventBuilder.build();
+        try {
+            eventOutput.write(event);
+        } catch (Exception e1) {
+            throw new DotRuntimeException(e1);
+        }
+
+        return eventOutput;
+    }
+
+
+    private class MyTailerListener extends TailerListenerAdapter {
+
+        StringWriter out = new StringWriter();
+
+
+        public void handle(String line) {
+            getOut().append(UtilMethods.xmlEscape(line) + "<br />");
+        }
+
+        StringWriter getOut() {
+            return getOut(false);
+        }
+
+        StringWriter getOut(boolean refresh) {
+            synchronized (this) {
+                if (refresh) {
+                    StringWriter s = new StringWriter().append(out.toString());
+                    this.out = new StringWriter();
+                    return s;
+                }
+                return out;
+            }
+        }
+
+    }
+
+    class MyTailerThread extends Thread {
+
+        Tailer tailer;
+        EventOutput eventOutput;
+        MyTailerListener listener;
+        String fileName;
+
+        @Override
+        public final void run() {
+
+            final OutboundEvent.Builder eventBuilder = new OutboundEvent.Builder();
+            try {
+                while (!eventOutput.isClosed()) {
+                    final String write = listener.getOut(true).toString();
+                    if (write != null && write.length() > 0) {
+                        eventBuilder.name("success");
+                        eventBuilder.data(Map.class,
+                                map("lines", write));
+                        eventBuilder.mediaType(MediaType.APPLICATION_JSON_TYPE);
+                        final OutboundEvent event = eventBuilder.build();
+                        eventOutput.write(event);
+                    }
+                    Thread.sleep(1000);
+                }
+            } catch (Exception ex) {
+                Logger.warn(this.getClass(), "Stopping listening log events for " + fileName + "Reason: " + ex.getMessage());
+            } finally {
+                stopTailer();
+                try {
+                    eventOutput.close();
```

Comment:
you can use CloseUtils.closeQuietly(eventOutput) to avoid the try catch

Reply:
Done